permuta/enumeration_strategies/insertion_encodable.py

```diff
@@ -1,13 +1,14 @@
-"""
-Enumeration strategies related to the insertion encoding.
-"""
+from typing import Iterable
 
+from permuta import Perm
 from permuta.enumeration_strategies.abstract_strategy import (
     EnumerationStrategyWithSymmetry,
 )
 from permuta.permutils.insertion_encodable import InsertionEncodablePerms
 
 
 class InsertionEncodingStrategy(EnumerationStrategyWithSymmetry):
-    def _applies_to_symmetry(self, b):
-        return InsertionEncodablePerms.is_insertion_encodable_maximum(b)
+    """Enumeration strategies related to the insertion encoding."""
+
+    def _applies_to_symmetry(self, basis: Iterable[Perm]) -> bool:
+        return InsertionEncodablePerms.is_insertion_encodable_maximum(basis)
```

Comment:
This strategy only needs to check two symmetries as it is a top to bottom style strategy. Therefore you only need to check the basis plus say its rotate 90 symmetry. Should we take advantage of that?

Reply:
Yeah. Noted. Will do in "next batch of refactoring"